Bottom: c66b7ff767484dd77cd251dbd412528bb8b2fc12
Top:    e55b1aac1a057c49e9200c70f69837dd0516a5aa
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-13 16:42:08 +0000

Calculate epsilon estimate for debugging


---

diff --git a/src/polygon.c b/src/polygon.c
index f488d79..17e4791 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -124,6 +124,7 @@ void
 polygon_init (void)
 {
   struct rlimit limit;
+  double eps;
 
   double cos_ang = cos (2.0 * M_PI / POLY_CIRC_SEGS_F);
   double sin_ang = sin (2.0 * M_PI / POLY_CIRC_SEGS_F);
@@ -139,6 +140,11 @@ polygon_init (void)
   limit.rlim_cur = MIN (limit.rlim_cur, 7000 * 1024 * 1024 /* 2000 GiB limit to virtual memory size */);
   setrlimit (RLIMIT_AS, &limit);
 
+  /* Estimate eps on this machine */
+  eps = 1.;
+  do eps = 0.5 * eps; while (eps + 1.0 > 1.0);
+
+  fprintf (stderr, "Epsilon calculated at %.17e\n", eps);
 }
 
 Cardinal
